kili/mutations/asset/__init__.py

```diff
@@ -98,43 +94,24 @@ def append_many_to_dataset(
             - For more detailed examples on how to import text assets,
                 see [the recipe](https://github.com/kili-technology/kili-python-sdk/blob/master/recipes/import_text_assets.ipynb).
         """
-        kili = QueriesProject(self.auth)
-        projects = kili.projects(project_id, disable_tqdm=True)
-        assert len(projects) == 1, NO_ACCESS_RIGHT
-        input_type = projects[0]["inputType"]
-        (properties_to_batch, upload_type, request,) = process_append_many_to_dataset_parameters(
-            input_type,
-            content_array,
-            external_id_array,
-            is_honeypot_array,
-            status_array,
-            json_content_array,
-            json_metadata_array,
-        )
 
-        def generate_variables(batch):
-            if request == GQL_APPEND_MANY_FRAMES_TO_DATASET:
-                payload_data = {
-                    "contentArray": batch["content_array"],
-                    "externalIDArray": batch["external_id_array"],
-                    "jsonMetadataArray": batch["json_metadata_array"],
-                    "uploadType": upload_type,
-                }
-            else:
-                payload_data = {
-                    "contentArray": batch["content_array"],
-                    "externalIDArray": batch["external_id_array"],
-                    "isHoneypotArray": batch["is_honeypot_array"],
-                    "statusArray": batch["status_array"],
-                    "jsonContentArray": batch["json_content_array"],
-                    "jsonMetadataArray": batch["json_metadata_array"],
-                }
-            return {"data": payload_data, "where": {"id": project_id}}
+        if content_array is None and json_content_array is None:
+            raise ValueError("Variables content_array and json_content_array cannot be both None.")
+        nb_data = len(content_array) if content_array is not None else len(json_content_array)
+
+        assets = [
+            {
+                "content": content_array and content_array[i],
+                "json_content": json_content_array and json_content_array[i],
+                "external_id": external_id_array and external_id_array[i],
+                "status": status_array and status_array[i],
+                "json_metadata": json_metadata_array and json_metadata_array[i],
+                "is_honeypot": is_honeypot_array and is_honeypot_array[i],
+            }
+            for i in range(nb_data)
+        ]
 
-        results = _mutate_from_paginated_call(
-            self, properties_to_batch, generate_variables, request
-        )
-        return format_result("data", results[0], Asset)
+        import_assets(self.auth, project_id=project_id, assets=assets)
```

Comment:
you can replace `self.auth` by self, so you don't need to instanciate kili again.

Reply:
but self is not `Kili` but `MutationAsset` here. So i won't be able to run other queries to get informations like the project `input_type` in `import_assets` because `projects` is a method of `Kili` and not `MutationAsset`
And in the new architecture, services won't call kili.projects but `get_input_type` (to code) that will not instantiate a new kili client, isn't it ?

Reply:
ok I did not see that. I think that each Mixin should have access to a client, since they don't have a reason to exist otherwise. To be discussed later then.